### PR TITLE
puppet cert sudo now optional

### PR DESCRIPTION
**Pull request by lzap**
REVIEW ONLY

How about this? Please someone test.

### CODE REVIEW

File: config/settings.yml.example [hunk 1]
```diff
@@ -74,6 +74,8 @@
 :puppetca: false
 #:ssldir: /var/lib/puppet/ssl
 #:puppetdir: /etc/puppet
+#:use_sudo: true
```
Comment (domcleal) [inline config/settings.yml.example]: What's the intended scope of this setting?

File: lib/proxy/puppetca.rb [hunk 2]
```diff
@@ -71,7 +71,7 @@ def autosign_list
     # list of all certificates and their state/fingerprint
     def list
       find_puppetca
-      command = "#{@sudo} -S #{@puppetca} --list --all"
```
Comment (domcleal) [inline lib/proxy/puppetca.rb:74]: Why removing -S?

File: lib/proxy/util.rb [hunk 3]
```diff
@@ -45,6 +45,12 @@ def self.join_all
     end
   end
 
+  # convert setting to boolean (with a default value)
+  def to_bool(value, default = false)
```
Comment (domcleal) [inline lib/proxy/util.rb:49]: Can you add tests?

File: config/settings.yml.example (hunk 1 above)
Comment (lzap) [inline config/settings.yml.example]: puppetca

I will rename to puppetca_use|_sudo.

File: lib/proxy/puppetca.rb (hunk 2 above)
Comment (lzap) [inline lib/proxy/puppetca.rb:74]: Because it's added elsewhere. Line 135.

**Comment by lzap** [comment]
Test added, rebased.

**Comment by Ginja** [comment]
@lzap, I've tested this and it resolves the issue I raised in #136. :+1: 

@domcleal, @lzap, any thoughts on which PR should be merged?

**Comment by domcleal** [comment]
@Ginja I'm going to take this one as it gives us more flexibility.  Thanks for testing and your earlier PR, much appreciated!

Merged as 3824d18 for Foreman 1.5.0, thanks @lzap!

**Pull request closed without merging**